ffi/src/transaction/mod.rs

```diff
@@ -192,16 +208,58 @@ mod tests {
     fn create_file_metadata(
         path: &str,
         num_rows: i64,
+        exclude_data_change: bool,
     ) -> Result<ArrowFFIData, Box<dyn std::error::Error>> {
-        let schema: ArrowSchema = add_files_schema().as_ref().try_into_arrow()?;
-
+        // Explicitly define the schema here which duplicates Transaction::add_files_schema, to demonstrate
+        // engines can still specify data change explicitly (i.e. preserve backwards compatibility for
+        // non-default engines).
+        let mut schema_fields = vec![
```

Comment:
Why not just add a static method on `Transaction` that can return the right schema depending on if you want data change or not, and call that, since you have both correct schemas statically in txn

Reply:
after discussing offline, i've removed backwads compatibility and now pass this schema through from the transaction `add_files_schema`